salt/minion.py

```diff
@@ -819,11 +819,11 @@ def __init__(self, opts):
                 penv = 'base'
             cache_top = {penv: {self.opts['id']: ['cache']}}
             with salt.utils.files.fopen(ptop, 'wb') as fp_:
-                salt.utils.yaml.safe_dump(cache_top, fp_)
+                salt.utils.yaml.safe_dump(cache_top, fp_, encoding='utf-8')
```

Comment:
What happens when the data is in fact in another encoding? Would it be better to use `__salt_system_encoding__` here?

Reply:
That's probably a good idea 👍

Reply:
I think we need too be careful about using `__salt_system_encoding__` here. There is a very good chance that will break on windows where the system encoding will be `cp1252`. In other places we try the system encoding then fall back to `UTF-8` if there is an exception. However in this case, since this is a cache file that is internal to salt, I actually think it's better to stick with `UTF-8`. Salt is the only thing reading/writing the file.

Reply:
@dwoz apparently I need to dig a little further, but I don't see that we're just reading the pillar data using a block like this, so it's very possible that we're using the system encoding to read the pillar data. I'm looking into that.